fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt

```diff
@@ -50,6 +51,7 @@ class NavigationToolbarTest {
     @After
     fun tearDown() {
         mockWebServer.shutdown()
+        AppAndSystemHelper.resetSystemLocaleToEnUS()
     }
```

Comment:
I'd like to start moving away from using `@After` and any `tearDown` functions and instead include everything in a `@Before` hook in a `setUp` function.

This will always run even if the runner or device crashes and the `@After` hook isn't called.

To enable this, there will need to me more checks in the `setUp`, but an example might be:

```
@Before
fun setUp() {
    # first run functions to cover a possible missed `@After` hook on runner crash

    # calling shutdown on a server that isn't running typically doesn't throw an error
    # but verify if this is the case as we might need to `catch` it
    mockWebServer.shutdown()
    AppAndSystemHelper.resetSystemLocaleToEnUS()
    # Include any other checks that need to replace calls in existing @After hooks
    ...
    
    # Now include your standard @Before hooks for your test setUp
    mockWebServer.start()
    # include anything else you need
    ...
}

Reply:
If implementing this pattern is difficult or impractical for any reason, then please let me know because it's an indication we need to modify our test `setup/teardown` strategy from a lower level that might require migrating some code and disconnecting some low-level hooks that, IMHO, aren't appropriate for UI testing enablement.

Reply:
I've done some test runs and noticed [Bug 1812183](https://bugzilla.mozilla.org/show_bug.cgi?id=1812183) when I switch the device language before the test starts. Due to the mentioned bug, for the language change to take effect, you need to restart the app. So the only way we can avoid the bug and also make sure we're resseting the language after the test crashes, the only way left is inside the @After methods.

Reply:
okay sounds good for now, thanks for the context.

there's a larger concern we'll have to find a good way to address, and that's implementing workarounds for known bugs.

when we do that, we lose the value of the measurement of a test in general, or at least dilute that value, by either requiring the test to be flaky if we know the first run will always fail, like for `SearchSuggest` or we run the risk of avoiding addressing the bug until later than we'd like. 

we'll figure it out later.